assets/source/setup-guide/app/steps/SetupTracking.js

```diff
@@ -206,6 +207,12 @@ const SetupTracking = ( { view } ) => {
 				await setAppSettings( {
 					[ name ]: value ?? ! appSettings[ name ],
 				} );
+				if (
+					appSettings?.tracking_advertiser &&
+					appSettings?.tracking_tag
+				) {
+					await connectAdvertiser();
```

Comment:
I'm not sure if I'm understanding things correctly here. We are calling `connectAdvertiser` in `saveOptions`, and `saveOptions` is called in `handleOptionChange`, so we are connecting advertiser in every option change? Is this expected?

Reply:
`handleOptionChange` is called each time that selected advertiser or tag change, since ads/v4 API requires to connect the advertiser with the tag, we need to verify that current combination of advertiser and tag is properly connected to the platform and reconnect it each time that selected values changes. The reconnect parameter in the response indicates that the combination changed or not. The remote request to the platform only is called if that combination is different than the previously connected one.

Reply:
Imho we need to verify if the current combination is connected or not every time that the selected options change

Reply:
@alexiglesias31 , thank you for the info. 🙂 

Just want to be sure, have you checked out the demo video I posted in https://github.com/woocommerce/pinterest-for-woocommerce/pull/305#issuecomment-999819703? I have only two advertiser accounts, I keep switching back and forth between them, and the notice "Advertiser connected successfully." would occasionally randomly show up for either one of them. Is this expected?

> we need to verify that current combination of advertiser and tag is properly connected to the platform and reconnect it each time that selected values changes.

I was expecting that verification and connection would only happen when users click on the "Complete Setup" button. Is that something that is not feasible, or perhaps invalid or out of scope? 🤔

Reply:
@ecgan we added a couple of modifications in order than connect advertiser only runs when users click the "Complete Setup" or "Save Settings" buttons.

Reply:
Perfect @ecgan. Merge conflict already solved 😉.